bundles/org.openhab.binding.ipobserver/src/main/java/org/openhab/binding/ipobserver/internal/IpObserverHandler.java

```diff
@@ -0,0 +1,314 @@
+            createChannelHandler(WIND_MAX_GUST, QuantityType.class, Units.METRE_PER_SECOND, "dailygust");
+        } else {
+            logger.warn(
+                    "The IP Observer is sending a wind format the binding does not support. Select one of the other units.");
+        }
+
+        if (config.solarUnit.equals("1")) {
+            createChannelHandler(SOLAR_RADIATION, QuantityType.class, Units.IRRADIANCE, "solarrad");
+        } else if (config.solarUnit.equals("0")) {
+            createChannelHandler(SOLAR_RADIATION, QuantityType.class, Units.LUX, "solarrad");
+        } else {
+            logger.warn(
+                    "The IP Observer is sending fc (Foot Candles) for the solar radiation. Select one of the other units.");
+        }
+
+        if (config.pressureUnit.equals("0")) {
+            createChannelHandler(ABS_PRESSURE, QuantityType.class, MetricPrefix.HECTO(SIUnits.PASCAL), "AbsPress");
+            createChannelHandler(REL_PRESSURE, QuantityType.class, MetricPrefix.HECTO(SIUnits.PASCAL), "RelPress");
+        } else if (config.pressureUnit.equals("1")) {
+            createChannelHandler(ABS_PRESSURE, QuantityType.class, ImperialUnits.INCH_OF_MERCURY, "AbsPress");
+            createChannelHandler(REL_PRESSURE, QuantityType.class, ImperialUnits.INCH_OF_MERCURY, "RelPress");
+        } else if (config.pressureUnit.equals("2")) {
+            createChannelHandler(ABS_PRESSURE, QuantityType.class, Units.MILLIMETRE_OF_MERCURY, "AbsPress");
+            createChannelHandler(REL_PRESSURE, QuantityType.class, Units.MILLIMETRE_OF_MERCURY, "RelPress");
+        }
+
+        createChannelHandler(WIND_DIRECTION, QuantityType.class, Units.DEGREE_ANGLE, "windir");
+        createChannelHandler(INDOOR_HUMIDITY, DecimalType.class, Units.PERCENT, "inHumi");
+        createChannelHandler(OUTDOOR_HUMIDITY, DecimalType.class, Units.PERCENT, "outHumi");
+        // The units for the following are ignored as they are not a QuantityType.class
+        createChannelHandler(UV, DecimalType.class, SIUnits.CELSIUS, "uv");
+        createChannelHandler(UV_INDEX, DecimalType.class, SIUnits.CELSIUS, "uvi");
+        // was outBattSta1 so some units may use this instead?
+        createChannelHandler(OUTDOOR_BATTERY, StringType.class, Units.PERCENT, "outBattSta");
+        createChannelHandler(OUTDOOR_BATTERY, StringType.class, Units.PERCENT, "outBattSta1");
+        createChannelHandler(INDOOR_BATTERY, StringType.class, Units.PERCENT, "inBattSta");
+        createChannelHandler(LAST_UPDATED_TIME, StringType.class, SIUnits.CELSIUS, "CurrTime");
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(IpObserverConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+        pollingFuture = scheduler.scheduleWithFixedDelay(this::pollStation, 1, config.pollTime, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        channelHandlers.clear();
+        @Nullable
```

Comment:
The compiler is more intelligent on local variables. Therefore, the Nullable could be removed.

Reply:
Thanks added this change.